larsim/GenericCRT/GenericCRT.cxx

```diff
@@ -0,0 +1,150 @@
+/*!
+ * Title:   GenericCRT Utility Class
+ * Author:  Andrzej Szelc (andrzejs@fnal.gov)
+ *
+ * Description:
+ * Class with Algorithms to convert sim::AuxDetHits to sim::AuxDetSimChannels
+ *
+*/
+
+
+#include "GenericCRT.h"
+
+#include <utility> // std::move()
+#include <algorithm> // std::find()
+
+
+
+
+// sim::GenericCRTUtility::GenericCRTUtility()
+// {
+//
+//
+// }
+
+
+
+unsigned int sim::GenericCRTUtility::CopyAuxDetHittoAuxDetIDE(const sim::AuxDetHit &InputHit,sim::AuxDetIDE &outputIDE)
+{
+
+
+   outputIDE.trackID		    = InputHit.GetTrackID();
+   outputIDE.energyDeposited	= InputHit.GetEnergyDeposited();
+   outputIDE.entryX		= InputHit.GetEntryX();
+   outputIDE.entryY		= InputHit.GetEntryY();
+   outputIDE.entryZ		= InputHit.GetEntryZ();
+   outputIDE.entryT		= InputHit.GetEntryT();
+   outputIDE.exitX		= InputHit.GetExitX();
+   outputIDE.exitY		= InputHit.GetExitY();
+   outputIDE.exitZ		= InputHit.GetExitZ();
+   outputIDE.exitT		= InputHit.GetExitT();
+   outputIDE.exitMomentumX	= InputHit.GetExitMomentumX();
+   outputIDE.exitMomentumY	= InputHit.GetExitMomentumY();
+   outputIDE.exitMomentumZ	= InputHit.GetExitMomentumZ();
+
+
+   return  InputHit.GetID();
```

Comment:
As mentioned in the header, I suggest rewriting this function as:

```c++
sim::AuxDetIDE sim::GenericCRTUtility::toAuxDetIDE(const sim::AuxDetHit &InputHit) const
{
  sim::AuxDetIDE outputIDE;
  // ...
  return outputIDE;
}
```

Reply:
Changes made